go/test/endtoend/onlineddl/flow/onlineddl_flow_test.go

```diff
@@ -0,0 +1,664 @@
+				assert.GreaterOrEqual(t, appliedDMLEnd-appliedDMLStart, int64(maxTableRows))
+				t.Logf("Applied DML: %d", appliedDMLEnd-appliedDMLStart)
+			})
+			t.Run("attempt to complete", func(t *testing.T) {
+				onlineddl.CheckCompleteMigration(t, &vtParams, shards, uuid, true)
+			})
+			isComplete := false
+			t.Run("optimistic wait for migration completion", func(t *testing.T) {
+				select {
+				case <-time.After(10 * time.Second):
+				case <-ctx.Done():
+					require.Fail(t, "context cancelled")
+				}
+
+				status := onlineddl.WaitForMigrationStatus(t, &vtParams, shards, uuid, migrationWaitTimeout, schema.OnlineDDLStatusRunning, schema.OnlineDDLStatusComplete)
+				isComplete = (status == schema.OnlineDDLStatusComplete)
+				fmt.Printf("# Migration status (for debug purposes): <%s>\n", status)
+			})
+			if !isComplete {
+				t.Run("force complete cut-over", func(t *testing.T) {
+					onlineddl.CheckForceMigrationCutOver(t, &vtParams, shards, uuid, true)
+				})
+				t.Run("another optimistic wait for migration completion", func(t *testing.T) {
+					status := onlineddl.WaitForMigrationStatus(t, &vtParams, shards, uuid, migrationWaitTimeout, schema.OnlineDDLStatusRunning, schema.OnlineDDLStatusComplete)
+					isComplete = (status == schema.OnlineDDLStatusComplete)
+					fmt.Printf("# Migration status (for debug purposes): <%s>\n", status)
+				})
+			}
+			if !isComplete {
+				t.Run("terminate workload", func(t *testing.T) {
+					// Seems like workload is too high and preventing migration from completing.
+					// We can't go on forever. It's nice to have normal completion under workload,
+					// but it's not strictly what this test is designed for. We terminate the
+					// workload so as to allow the migration to complete.
+					cancelWorkload()
+				})
+			}
+			t.Run("wait for migration completion", func(t *testing.T) {
+				status := onlineddl.WaitForMigrationStatus(t, &vtParams, shards, uuid, migrationWaitTimeout, schema.OnlineDDLStatusComplete)
+				fmt.Printf("# Migration status (for debug purposes): <%s>\n", status)
+				onlineddl.CheckMigrationStatus(t, &vtParams, shards, uuid, schema.OnlineDDLStatusComplete)
+			})
+			t.Run("validate table schema", func(t *testing.T) {
+				checkMigratedTable(t, tableName, hint)
+			})
+
+			cancelWorkload() // Early break
+			cancel()         // Early break
+			wg.Wait()
+		})
+	})
+}
+
+func testWithInitialSchema(t *testing.T) {
+	// Create the stress table
+	err := clusterInstance.VtctldClientProcess.ApplySchema(keyspaceName, createStatement)
+	require.Nil(t, err)
+
+	// Check if table is created
+	checkTable(t, tableName)
+}
+
+// testOnlineDDLStatement runs an online DDL, ALTER statement
+func testOnlineDDLStatement(t *testing.T, alterStatement string, ddlStrategy string, expectHint string, skipWait bool) (uuid string) {
+	row := onlineddl.VtgateExecDDL(t, &vtParams, ddlStrategy, alterStatement, "").Named().Row()
+	if row != nil {
+		uuid = row.AsString("uuid", "")
+	}
```

Comment:
Should we not use require/assert to fail here if we don't get a row and/or uuid?

Reply:
Yes. This is copy+paste from other tests where a `nil` result is in fact possible (where we expect an error in submission).

Reply:
fixed